rift/node.py

```diff
@@ -2469,87 +2517,116 @@ def spf_run_direction(self, spf_direction):
         # candidate heap, it (potentially) updates the cost and predecessors of the neighbor.
         #
         # Consider each neighbor of the visited node in the direction of the SPF
-        for nbr in self.node_neighbors(node_ties, spf_direction):
-            (nbr_system_id, nbr_tie_element) = nbr
+
+        # If the special_for_neg_disagg = True, we add E-W neighbors
+        directions = [spf_direction] if not special_for_neg_disagg else [spf_direction, constants.DIR_EAST_WEST]
+
+        for nbr_system_id, nbr_tie_element in self.node_neighbors(node_ties, neighbor_directions=directions):
             # Only consider bi-directional adjacencies.
             if self.is_neighbor_bidirectional(node_system_id, nbr_system_id, nbr_tie_element,
-                                              spf_direction):
+                                              spf_direction, special_for_neg_disagg):
                 # We have found a feasible path to the neighbor node; is the best path?
                 cost = node_cost + nbr_tie_element.cost
                 destination = spf_dest.make_node_dest(nbr_system_id, None, cost)
+
                 self.spf_consider_candidate_dest(destination, nbr_tie_element, node_system_id,
-                                                 candidates, spf_direction)
+                                                 candidates, spf_direction, special_for_neg_disagg)
+
+    def spf_add_prefixes(self, node_sysid, node_cost, candidates, spf_direction, special_for_neg_disagg):
+        prefix_ties = self.ties_of_type(direction=self.spf_use_tie_direction(node_sysid, spf_direction),
+                                        system_id=node_sysid,
+                                        prefix_type=common.ttypes.TIETypeType.PrefixTIEType
+                                        )
 
-    def spf_add_prefixes(self, node_sysid, node_cost, candidates, spf_direction):
-        prefix_ties = self.ties_of_type(
-            direction=self.spf_use_tie_direction(node_sysid, spf_direction),
-            system_id=node_sysid,
-            prefix_type=common.ttypes.TIETypeType.PrefixTIEType)
         for prefix_tie in prefix_ties:
-            self.spf_add_prefixes_common(
-                node_sysid, node_cost, candidates, spf_direction,
-                prefix_tie.element.prefixes.prefixes, False)
-
-    def spf_add_pos_disagg_prefixes(self, node_sysid, node_cost, candidates, spf_direction):
-        prefix_ties = self.ties_of_type(
-            direction=self.spf_use_tie_direction(node_sysid, spf_direction),
-            system_id=node_sysid,
-            prefix_type=common.ttypes.TIETypeType.PositiveDisaggregationPrefixTIEType)
+            self.spf_add_prefixes_common(node_sysid, node_cost, candidates, spf_direction, special_for_neg_disagg,
+                                         prefix_tie.element.prefixes.prefixes,
+                                         is_pos_disagg=False
+                                         )
```

Comment:
Don't put closing bracket on separate line. Keep putting arguments on same line, as long as right margin of 100 is not reached.